### PR TITLE
Contact Form Block: Fix Missing Button Color Attributes

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request:

The Contact Form block was missing the `textButtonColor` and `backgroundButtonColor` attributes, necessary to make it work with the `SubmitButton` shared component.
This practically means that, currently, you can change the colors of the Contact Form submit button, you will see the button preview updating correctly (as the colors are applied as CSS classes in the editor), but if you save the post and reload, the colors are not saved.
This only happens for "named" colors; custom colors work fine, as their attributes are not missing.

This PR adds those missing two attributes to the Contact Form block.
Also, externalizes the `colourValidator` utility that was being carelessly copypasted around.
I've also americanized the function name, as it's more consistent with the fact that we use `color` everywhere, and I'm ready to fight my British colleagues about it! 😄 

Note: most changed lines here are automatic Prettier changes on the test file. 😭 

#### Testing instructions:

* Insert a Contact Form block.
* Modify the button colors, picking from the named colors palette.
* Make sure the button updates correctly.
* Save and reload the editor.
* Make sure the button shows up with the correct colors.
* Try again with custom colors, and make sure there are no regressions.

#### Proposed changelog entry for your changes:

* Contact Form Block: Fix button colors not saving correctly.

### CODE REVIEW

File: extensions/shared/colorValidator.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const hexRegex = /^#?[A-Fa-f0-9]{6}$/;
+
+export default function colourValidator( value ) {
```
Comment (scruffian) [inline extensions/shared/colorValidator.js]: Well if you're going to Americanize the name you should at least do it here too!

**Review by scruffian — DISMISSED** [review]
This looks good, just one nitpick about spelling!

File: extensions/shared/colorValidator.js (hunk 1 above)
Comment (Copons) [reply extensions/shared/colorValidator.js]: HA!

**Review by jeherve — APPROVED** [review]
Looks good in my tests. 👍

**Comment by Copons** [comment]
r203932-wpcom

**Comment by pablinos** [comment]
Good catch! It looks like we didn't pick up the Contact Form [with this change](https://github.com/Automattic/jetpack/pull/14599) which is probably because it doesn't explicitly use the SubmitButton component.

AFAIK we shouldn't be validating the colours has hex values though. I had to remove that validation from another block as the colours can be values like `light-gray` coming from the theme pallette. I think this may be contributing to the problems we're seeing in [the AMP change ](https://github.com/Automattic/jetpack/pull/14630)

**Comment by Copons** [comment]
@pablinos I think you're right: we should only validate _custom_ colors (that are supposed to always be hex codes), but not _named_ ones.
To be honest, I've done that on Contact Form, and simply updated all the other `colorValidator` uses with the new shared utility, without thinking much about it.

**Comment by pablinos** [comment]
Yeah looking at it again, I think I got confused by the all the colour related attributes for the Calendly block. I think there might be a couple we can deprecate now.

**Pull request merged**